tests/lib/rule-tester/rule-tester.js

```diff
@@ -2042,11 +2042,15 @@ describe("RuleTester", () => {
     // Nominal message/messageId use cases
     it("should assert match if message provided in both test and result.", () => {
         assert.throws(() => {
-            ruleTester.run("foo", require("../../fixtures/testers/rule-tester/messageId").withMessageOnly, {
-                valid: [],
-                invalid: [{ code: "foo", errors: [{ message: "something" }] }]
-            });
-        }, /Avoid using variables named/u);
+            try {
+                ruleTester.run("foo", require("../../fixtures/testers/rule-tester/messageId").withMessageOnly, {
+                    valid: [],
+                    invalid: [{ code: "foo", errors: [{ message: "something" }] }]
+                });
+            } catch (error) {
+                throw new Error(`Expect message to be "${error.expected}", got "${error.actual}"`);
+            }
+        }, 'Expect message to be "something", got "Avoid using variables named \'foo\'."');
```

Comment:
I'm not sure why `assert.strictEqual()` produces errors with the diff included in `message` (per https://nodejs.org/api/assert.html#strict-assertion-mode, that should happen only in strict assertion mode? But we're not using strict assertion mode). Either way, this assertion was not very precise and it's fine to change it, especially if it's causing problems in some environments. We can just simplify it using an existing helper:

```suggestion
            ruleTester.run("foo", require("../../fixtures/testers/rule-tester/messageId").withMessageOnly, {
                valid: [],
                invalid: [{ code: "foo", errors: [{ message: "something" }] }]
            });
        }, assertErrorMatches("Avoid using variables named 'foo'.", "something"));
```

Reply:
Maybe caused by this line https://github.com/nodejs/node/blob/c3ed292d17c34578fd7806cb42da82bbe0cca103/lib/internal/assert/myers_diff.js#L152C44-L152C54

`colors.hasColors`